### PR TITLE
Fixed #10785: Modified berkowitz_eigenvals() function

### LINKED ISSUE #10785: Problem with berkowitz_eigenvals() with empty matrices

### PR DESCRIPTION
**Pull request by aravindkanna**
This fixes the issue [10785](https://github.com/sympy/sympy/issues/10785).

### CODE REVIEW

**Comment by aravindkanna** [comment]
As the empty matrix have no minors, I think it would be better to return an empty tuple for the `berkowitz_minors()` function.

**Comment by aravindkanna** [comment]
`In [87]: a.berkowitz_eigenvals()`
`Out[87]: {}`
`In [88]: a.berkowitz_minors()`
`Out[88]: ()`

Now this would be the output for two modified functions(a is an Empty Matrix).

**Comment by aravindkanna** [comment]
Also I would like to prefer to return `(1,)` as output for the `berkowitz()` function. Right now it is returning the same Index out of range error. 
And the reason being, what I thought, the characteristic polynomial of an empty matrix is one. Say A is empty matrix and I be identity matrix of shape (0, 0). Now `det(lamda*I - A)` is nothing but `det(A)` . That means the characteristic polynomial is a constant polynomial for empty matrices and the coefficient will be 1. So it is reasonable to return a tuple with 1 inside. i.e `(1,)`.

Please provide your inputs.

**Comment by aravindkanna** [comment]
@moorepants , sir please review this

**Comment by aravindkanna** [comment]
```No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.```
what is this?

**Comment by leosartaj** [comment]
240 commits seem to be a lot. Can you update this with master? So only latest commits are shown.

**Comment by aravindkanna** [comment]
Sir, what should I do now?

**Comment by aravindkanna** [comment]
@leosartaj , thank you :smile:

**Comment by aravindkanna** [comment]
@leosartaj , thank you :smile: can this be in?

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -3015,6 +3015,8 @@ def berkowitz_minors(self):
 
         berkowitz
         """
+        if not self:
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: Is this test necessary after ``berkowitz`` has been modified to work with an empty matrix?

**Comment by aravindkanna** [comment]
yeah, sorry. It is not necessary. I have created this before that. Thank you @jksuom . Can I close this now?

**Comment by jksuom** [comment]
It is not necessary to totally close the PR, the tests could remain.
This is what I get with the current master for ``M = Matrix()``:
```
In [3]: M.berkowitz_minors()
Out[3]: (1,)
In [4]: M.berkowitz_eigenvals()
Out[4]: {}
```
I think you could leave these tests in this PR (with the first one properly edited). On the other hand, there seems to be no reason for special treatment of empty matrices in ``berkowitz_minors`` and ``berkowitz_eigenvals``.
So the contents of this PR could consist of those tests alone.

**Comment by jksuom** [comment]
Looks good, thanks.

**Pull request merged**